openedx/frontend-app-course-authoring
# feat: better api error handling

## Description

Saves API errors in redux state and displays them as alerts.

## Supporting information
`Private-ref`: [BB-8831](https://tasks.opencraft.com/browse/BB-8831)

## Testing instructions

* Try blocking course outline index (`http://studio.local.edly.io:8001/api/contentstore/v1/course_index/course-v1:OpenedX+DemoX+DemoCourse` in tutor) fetch api using network tab in developer console.
* Refresh and you should see an alert: `Network error`.
* You can also update this [response](https://github.com/openedx/edx-platform/blob/45cd459ddec2ca73d16fc00dfcd2feaccb16efa6/cms/djangoapps/contentstore/rest_api/v1/views/course_index.py#L100-L100) in edx-platform and return a error status code with some message, it will be displayed in the alert. `return Response("some error", status=500)`

![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/15c1bcf7-4bda-4a2c-80f1-103a658a8007)


![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/cf3c3244-c339-4935-b09f-2580f728f086)
